### PR TITLE
Add delete permission to group and membership tables

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

**Comment by jacksonh** [comment]
classic

File: packages/api/src/services/groups.ts [hunk 1]
```diff
@@ -172,19 +172,25 @@ export const leaveGroup = async (
...
-        (m) => m.isAdmin && m.user.id !== user.id
+        (m) => m.isAdmin && m.id !== membershipId
       )
       if (!hasAdmin) {
         const newAdmin = group.members.find((m) => !m.isAdmin)
         if (!newAdmin) {
-          // delete the group if there are no more members
-          await t.getRepository(Group).delete({ id: group.id })
-          return true
+          throw new Error('No user found')
```
Comment (jacksonh) [inline packages/api/src/services/groups.ts:193]: So it wont delete the group anymore? (That is ok with me for now)

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/services/groups.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/services/groups.ts:193]: It will still delete the group (I moved the code to the other place)

Comment (sywhb) [reply packages/api/src/services/groups.ts:193]: @jacksonh

File: packages/api/src/services/groups.ts [hunk 2]
```diff
@@ -172,19 +172,25 @@ export const leaveGroup = async (
       throw new Error('User not in group')
     }
 
+    const membershipId = membership.id
     await t.getRepository(GroupMembership).remove(membership)
 
+    if (group.members.length === 1) {
```
Comment (jacksonh) [inline packages/api/src/services/groups.ts:178]: Ah i see it!

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/services/groups.ts [hunk 3]
```diff
@@ -172,19 +172,25 @@ export const leaveGroup = async (
...
+      // delete the group if there are no more members
+      await t.getRepository(Group).remove(group)
+
+      return true
+    }
+
```
Comment (sywhb) [inline packages/api/src/services/groups.ts:184]: I move the deletion of group here to fix a bug @jacksonh

**Pull request merged**